gtda/diagrams/_utils.py

```diff
@@ -59,6 +59,15 @@ def _filter(Xs, filtered_homology_dimensions, cutoff):
     return Xf
 
 
+def _differentiate(Xs):
```

Comment:
I am not sure "differentiate" is the right name for this.
How does this relate to a differential operator?

Reply:
It is similar to np.diff(, axis = 2). Since, for points of the form (b,d,q), we want (b,d-b,q), I could use np.diff, with post-diff concatenation ( * ),I thought it would be clearer to write it this way.
( * ) for example: `np.concatenate([X[:,:,0:1], np.diff(X, axis = 2)[:, :, 0:1], X[:,:,2:3]], axis = 2)`